NSubstitute.DbConnection/NSubstitute.DbConnection/DbConnectionWrapper.cs

```diff
@@ -35,10 +37,14 @@ public override string ConnectionString
 
         public override void Open() => _inner.Open();
 
+        public override async Task OpenAsync(CancellationToken cancellationToken) => _inner.Open();
+
         public DbDataReader ExecuteReader(IDbCommand mockCommand) => GetMatchingQuery(mockCommand).ExecuteReader(mockCommand);
 
         public int ExecuteNonQuery(IDbCommand mockCommand) => GetMatchingQuery(mockCommand).ExecuteNonQuery(mockCommand);
 
+        public object ExecuteScalar(IDbCommand mockCommand) => GetMatchingQuery(mockCommand).ExecuteScalar(mockCommand);
```

Comment:
Is this used? Or rather what is this (and the other execute methods) on here used for? They're not part of interface/base class

Reply:
Yes, these three methods are used by `DbConnectionMockExtensions.CreateMockCommand()` and they bind in the mock query behaviour for the Query, NonQuery and now Scalar variants of how you execute a query. There are various async overloads but they all just call the same synchronous method